From 1f16696d9fb83a5e255f7d8f96787ee6903b6404 Mon Sep 17 00:00:00 2001
From: Thor Thayer <thor.thayer@linux.intel.com>
Date: Fri, 11 Oct 2019 13:08:43 -0500
Subject: [PATCH 042/151] arm64: smp: Fix EDAC DBE panic_smp_self_stop()
 conflict

commit  87b2088d95754e6e6cfec7f9d6180ae59cb4af02 from
https://github.com/altera-opensource/linux-socfpga.git
branch is socfpga-5.4.64-lts

The EDAC Warm Reset on peripheral DBE now has a conflict
with the platform function which was recently added.

Replace the platform function with the EDAC version
when WARM_RESET is configured.

Signed-off-by: Thor Thayer <thor.thayer@linux.intel.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
Integrated-by: Jun Zhang <jun.zhang@windriver.com>
---
 arch/arm64/kernel/smp.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm64/kernel/smp.c b/arch/arm64/kernel/smp.c
index e196b171b984..20cf294273ea 100644
--- a/arch/arm64/kernel/smp.c
+++ b/arch/arm64/kernel/smp.c
@@ -845,6 +845,9 @@ static void local_cpu_stop(void)
 	cpu_park_loop();
 }
 
+#ifndef CONFIG_EDAC_ALTERA_ARM64_WARM_RESET
+/* In ECC_DBE_WARM_RESET case, use EDAC panic_smp_self_stop() */
+
 /*
  * We need to implement panic_smp_self_stop() for parallel panic() calls, so
  * that cpu_online_mask gets correctly updated and smp_send_stop() can skip
@@ -854,6 +857,7 @@ void panic_smp_self_stop(void)
 {
 	local_cpu_stop();
 }
+#endif
 
 #ifdef CONFIG_KEXEC_CORE
 static atomic_t waiting_for_crash_ipi = ATOMIC_INIT(0);
-- 
2.26.1

